module/mempool/queue/heroQueue.go

```diff
@@ -0,0 +1,71 @@
+package queue
+
+import (
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	herocache "github.com/onflow/flow-go/module/mempool/herocache/backdata"
+	"github.com/onflow/flow-go/module/mempool/herocache/backdata/heropool"
+)
+
+// HeroQueue implements a HeroCache-based in-memory queue.
```

Comment:
It's better to also mention briefly the feature of HeroCache, such as "a key-value cache with zero heap allocation"